Sources/UIScreen.swift

```diff
@@ -70,12 +70,20 @@ public final class UIScreen {
         SDL_Init(SDL_INIT_VIDEO | SDL_INIT_EVENTS)
         GPU_SetPreInitFlags(GPU_INIT_DISABLE_VSYNC)
 
+        #if os(Android)
+        let androidScreenDimension = getAndroidScreenDimension()
+        #endif
+
         if options.contains(SDL_WINDOW_FULLSCREEN), let displayMode = SDLDisplayMode.current {
             // Fix fullscreen resolution on Mac and make Android easier to reason about:
             // There is an inconsistency between Mac and Android when setting SDL_WINDOW_FULLSCREEN
             // The easiest solution is just to work in 1:1 pixels
             GPU_SetPreInitFlags(GPU_GetPreInitFlags() | GPU_INIT_DISABLE_AUTO_VIRTUAL_RESOLUTION)
+            #if os(Android)
+            size = CGSize(width: CGFloat(androidScreenDimension.width), height: CGFloat(androidScreenDimension.height))
+            #else
             size = CGSize(width: CGFloat(displayMode.w), height: CGFloat(displayMode.h))
```

Comment:
these dimensions here from `SDLDisplayMode.current` are wrong in that specific case. It seems that the displayMode caches the previous dimension and I did not find a way to reset it. We do properly deinit UIScreen, which calls GPU_quit

Reply:
I thought that we set `SDLDisplayMode.current` ourselves somewhere. In that case would an option be to update it on orientation change rather than working around the issue?

Reply:
By "we" I mean some code that we have control over, which is "everything", for better or worse 😅 I am pretty sure we could fix this at the source, but this workaround looks fine to me

Reply:
Agreed!